src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -32,62 +32,33 @@ function LinkIcon({ url }: { url: string }) {
   return <EuiImage size="s" alt="" aria-hidden={true} url={url} />;
 }
 
-export interface NavLink {
-  key: string;
-  label: string;
-  href: string;
-  isActive: boolean;
-  onClick(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
-  category?: AppCategory;
-  isDisabled?: boolean;
-  iconType?: string;
-  icon?: JSX.Element;
-  order?: number;
-  'data-test-subj': string;
+interface Props {
+  link: ChromeNavLink;
+  legacyMode: boolean;
+  appId: string | undefined;
+  basePath?: HttpStart['basePath'];
+  dataTestSubj: string;
+  onClick?: Function;
+  navigateToApp: CoreStart['application']['navigateToApp'];
 }
 
-/**
- * Create a link that's actually ready to be passed into EUI
- *
- * @param navLink
- * @param legacyMode
- * @param currentAppId
- * @param basePath
- * @param navigateToApp
- */
-export function createNavLink(
-  navLink: ChromeNavLink,
-  legacyMode: boolean,
-  currentAppId: string | undefined,
-  basePath: HttpStart['basePath'],
-  navigateToApp: CoreStart['application']['navigateToApp']
-): NavLink {
-  const {
-    legacy,
-    url,
-    active,
-    baseUrl,
-    id,
-    title,
-    disabled,
-    euiIconType,
-    icon,
-    category,
-    order,
-    tooltip,
-  } = navLink;
-  let href = navLink.url ?? navLink.baseUrl;
-
-  if (legacy) {
-    href = url && !active ? url : baseUrl;
-  }
+export function createEuiListItem({
+  link,
+  legacyMode,
+  appId,
+  basePath,
+  onClick = () => {},
+  navigateToApp,
+  dataTestSubj,
+}: Props) {
+  const { legacy, active, id, title, disabled, euiIconType, icon, tooltip, href } = link;
 
   return {
-    category,
-    key: id,
     label: tooltip ?? title,
-    href, // Use href and onClick to support "open in new tab" and SPA navigation in the same link
-    onClick(event) {
+    href,
+    /* Use href and onClick to support "open in new tab" and SPA navigation in the same link */
+    onClick(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
```

Comment:
NIT: Unnecessary change (that the second time someone want to make this signature verbose... What IDE are you guys using?)

<img width="663" alt="Screenshot 2020-05-15 at 23 26 52" src="https://user-images.githubusercontent.com/1532934/82414967-21150480-9a78-11ea-9ac5-08a92c4894c3.png">

The function returns a `NavLink`. `NavLink.onClick` is already defined as `(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void` So the inline `onClick` type is already properly inferred from TS.

Reply:
👍 Will remove the beast of a type definition. 

At some point during refactoring, it was complaining about it but once everything settled I guess TS can handle it now.

Reply:
🤔 VS Code does not like removing that type and `node scripts/type_check` fails as well. 

Looks like _your_ IDE is doing some magic.

Reply:
it's just delegating to the TS binary.

The file is passing TS typecheck on master, what change in the PR would cause it to now fail (and requires the explicit definition)?

Reply:
This function now executes in a slightly different part of the chain now (literally right before being passed into EUI now) so maybe TS can't keep track of it all the way through the transforms nav links take? Or maybe, the opposite, it was wrong before and now that TS can keep track of it all the way through...

My guess boils down to "typechecking isn't perfect" though. Not sure how to better investigate this.

Reply:
Ok. Just wanted to make sur the change was necessary. If it is, it's fine to keep it.